build.assets/robotest/run.sh

```diff
@@ -3,55 +3,114 @@ set -o errexit
-export INSTALLER_URL=$GRAVITY_BUILDDIR/robotest.tar
-export GRAVITY_URL=$GRAVITY_BUILDDIR/gravity
-export TAG=$(git rev-parse --short HEAD)
+readonly TARGET=${1:?Usage: [path to config]}
+
+GRAVITY_URL=${GRAVITY_URL:?Set GRAVITY_URL to the current gravity binary}
+INSTALLER_URL=${INSTALLER_URL:?Set INSTALLER_URL to the container local default robotest image}
+OPSCENTER_URL=${OPSCENTER_URL:?Set OPSCENTER_URL to the container local opscenter image}
+ROBOTEST_IMAGES_DIR=${ROBOTEST_IMAGES_DIR:? Set ROBOTEST_IMAGES_DIR to the directory with robotest images}
+STATEDIR=${STATEDIR:?Set STATEDIR to a suitable place to store robotest terraform state and logs}
+
+TAG=$(git rev-parse --short HEAD)
+
 # cloud provider that test clusters will be provisioned on
 # see https://github.com/gravitational/robotest/blob/v2.0.0/infra/gravity/config.go#L72
-export DEPLOY_TO=${DEPLOY_TO:-gce}
-export GCL_PROJECT_ID=${GCL_PROJECT_ID:-"kubeadm-167321"}
-export GCE_REGION="northamerica-northeast1,us-west1,us-east1,us-east4,us-central1"
+DEPLOY_TO=${DEPLOY_TO:-gce}
+GCL_PROJECT_ID=${GCL_PROJECT_ID:-"kubeadm-167321"}
+GCE_REGION="northamerica-northeast1,us-west1,us-east1,us-east4,us-central1"
 # GCE_VM tuned down from the Robotest's 7 cpu default in 09cec0e49e9d51c3603950209cec3c26dfe0e66b
 # We should consider changing Robotest's default so that we can drop the override here. -- 2019-04 walt
-export GCE_VM=${GCE_VM:-custom-4-8192}
+GCE_VM=${GCE_VM:-custom-4-8192}
+GCE_PREEMPTIBLE=${GCE_PREEMPTIBLE:-'false'}
 # Parallelism & retry, tuned for GCE
-export PARALLEL_TESTS=${PARALLEL_TESTS:-4}
-export REPEAT_TESTS=${REPEAT_TESTS:-1}
-
-# set SUITE and UPGRADE_VERSIONS
-case $TARGET in
-  pr) source $(dirname $0)/pr_config.sh;;
-  nightly) source $(dirname $0)/nightly_config.sh;;
-  *) echo "Unknown target $TARGET\nUsage: $0 [pr|nightly] [upgrade_from_dir]"; exit 1;;
-esac
-
-function build_volume_mounts {
-  for release in ${UPGRADE_VERSIONS[@]}; do
-      local tarball=$(tag_to_tarball ${release})
-      echo "-v $UPGRADE_FROM_DIR/$tarball:/$tarball"
-  done
+PARALLEL_TESTS=${PARALLEL_TESTS:-4}
+REPEAT_TESTS=${REPEAT_TESTS:-1}
+RETRIES=${RETRIES:-3}
+FAIL_FAST=${FAIL_FAST:-false}
+ALWAYS_COLLECT_LOGS=${ALWAYS_COLLECT_LOGS:-true}
+
+# what should happen with provisioned VMs on individual test success or failure
+DESTROY_ON_SUCCESS=${DESTROY_ON_SUCCESS:-true}
+DESTROY_ON_FAILURE=${DESTROY_ON_FAILURE:-true}
+
+# set SUITE and ROBOTEST_IMAGE_DIR_MOUNTPOINT
+export INSTALLER_URL OPSCENTER_URL
+source $TARGET
+
+# ROBOTEST_IMAGE_DIR_MOUNTPOINT defined by the config
+EXTRA_VOLUME_MOUNTS="-v $ROBOTEST_IMAGES_DIR:$ROBOTEST_IMAGE_DIR_MOUNTPOINT"
+
+
+# GRAVTIY_FILE/GRAVITY_URL specify the location of the up-to-date gravity binary
+if [ -d $(dirname ${GRAVITY_URL}) ]; then
+  GRAVITY_FILE='/robotest-bin/'$(basename ${GRAVITY_URL})
+  EXTRA_VOLUME_MOUNTS=${EXTRA_VOLUME_MOUNTS:-}" -v "$(dirname ${GRAVITY_URL}):$(dirname ${GRAVITY_FILE})
+fi
```

Comment:
File an enhancement bug to remove this dependency.  It should be quite unnecessary in this age.

Reply:
https://github.com/gravitational/robotest/issues/232